base_technical_user/models/models.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Akretion (http://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import models
+from odoo.exceptions import UserError
+
+
+def sudo_tech(self, raise_if_missing=False):
+    self_sudoer = self
+    tech_user = self.env.user.company_id.user_tech_id
+    if tech_user:
+        self_sudoer = self.sudo(tech_user.id)
+    elif raise_if_missing:
+        raise UserError("The technical user is missing in the company {}".format(self.env.user.company_id.name))
+    return self_sudoer
+
+
+models.BaseModel.sudo_tech = sudo_tech
```

Comment:
Why not doing like this:

```
class Base(models.AbstractModel):
    _inherit = 'base'

    def sudo_tech(self, raise_if_missing=False):
        self_sudoer = self
        tech_user = self.env.user.company_id.user_tech_id
        if tech_user:
            self_sudoer = self.sudo(tech_user.id)
        elif raise_if_missing:
            raise UserError("The technical user is missing in the company {}".format(self.env.user.company_id.name))
        return self_sudoer
```
It's more "Odoo"?

Reply:
Sorry I mix the version. I was think that this was not possible on 10 and only work since 11. But it's work since 9. I fix it